src/tests/ftest/control/dmg_storage_query.py

```diff
@@ -0,0 +1,198 @@
+        """Check if VOS ids provided by dmg match config value.
+
+        Args:
+            attr_name (str): Name of config attribute wanted.
+            expected_len (int): Len of the list that contains the config_attr.
+        """
+        params = self.server_managers[-1].runner.job.yaml_params
+        cfg_attr = getattr(params.server_params[-1], attr_name)
+        if expected_len != cfg_attr.value:
+            self.fail("Number of {}: {} dont match config val: {}".format(
+                attr_name, expected_len, cfg_attr.value))
+        return True
+
+    def check_smd_out(self, smd_info, valid_info, invalid_info):
+        """Check the smd output with specified
+
+        Args:
+            valid_info (str): Info from "devices" or "pools"
+            invalid_info (str): Info from "devices" or "pools"
+        """
+        if smd_info:
+            if valid_info in smd_info:
+                self.log.info("Found uuid: %s", smd_info[valid_info]["uuid"])
+                self.check_len("targets", len(smd_info[valid_info]["vos_ids"]))
+                # Perform additional check for pools uuid and blobs
+                if valid_info == "pools":
+                    if smd_info[valid_info]["uuid"] != self.pool.uuid:
+                        self.fail("Pool uuids don't match: {} {}".format(
+                            smd_info[valid_info]["uuid"], self.pool.uuid))
+                    self.check_len(
+                        "targets", len(smd_info[valid_info]["spdk_blobs"]))
+            else:
+                self.fail("Wanted information not found: {}".format(smd_info))
+
+            if invalid_info in smd_info:
+                self.fail("Found unwanted pool information: {}".format(
+                    smd_info[invalid_info]))
+        else:
+            self.fail("SMD info not found: {}".format(smd_info))
+
+    def check_dev_state_out(self, smd_info, method_name, state):
+        """Check device state output.
+
+        Args:
+            state (str): State expected. e.i. NORMAL | FAULTY
+        """
+        # Parse stdout into a dict check that devices information is displayed.
+        device_state_info = []
+        for info in smd_info:
+            if "Device" in info:
+                uuid = info[info.index("Device") + 1]
+                device_state_info.append(
+                    self.get_dmg_cmd_info(method_name, uuid))
+
+        # Check that the state of each device is NORMAL
+        for dev in device_state_info:
+            if dev[2] != state:
+                self.fail("Found a device in {} state.".format(dev[2]))
+
+    def test_dmg_storage_query_smd_devices(self):
+        """
+        JIRA ID: DAOS-3925
+
+        Test Description: Test 'dmg storage query smd -d' command.
+
+        :avocado: tags=all,pr,hw,small,dmg_storage_query,smd_devs,basic
+        """
+        # Get the storage smd infromation, parse and check devices info
+        smd_info = self.get_dmg_cmd_info("storage_query_smd", pools=False)
```

Comment:
(pylint-unexpected-keyword-arg) Unexpected keyword argument 'pools' in method call
(pylint-no-value-for-parameter) No value for argument 'kwargs' in method call